infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -310,6 +336,53 @@ private String getLocalEndpoint() {
     }
 
     /**
+     * Task to collect local node metrics.
+     * This task collects the following:
+     *  Boolean Status of layout, sequencer and logunit servers.
+     *  Logunit Metrics:
+     *      Size of writeQueue for pending write requests.
+     * These metrics are then composed into NodeMetrics protobuf model
+     * and stored locally.
+     */
+    private void localMetricsPollingScheduler() {
```

Comment:
I will update the name.
Yes you are right, we do not want simultaneous access.
I can use a concurrent hashmap for simultaneous writes.
For reads only, we can return an immutable copy of the map.